examples/bsoncxx/builder_basic.cpp

```diff
@@ -0,0 +1,60 @@
+#include <bsoncxx/builder/basic/array.hpp>
+#include <bsoncxx/builder/basic/document.hpp>
+#include <bsoncxx/builder/basic/kvp.hpp>
+#include <bsoncxx/types.hpp>
+
+using namespace bsoncxx;
+
+int main(int, char**) {
+
+    // bsoncxx::builder::basic presents a BSON-construction interface familiar to users of the server's
+    // BSON library or the Java driver.
+
+    // basic::document builds a BSON document.
+    auto doc = builder::basic::document{};
+    // basic::array builds a BSON array
+    auto arr = builder::basic::array{};
+
+    // We append key-value pairs to a document using the kvp helper.
+    using bsoncxx::builder::basic::kvp;
+
+    doc.append(kvp("foo", "bar")); // string literal value will be converted to b_utf8 automatically
+    doc.append(kvp("baz", types::b_bool{false}));
+    doc.append(kvp("garply", types::b_double{3.14159}));
+
+    // We can also pass a variable number of keys to append.
```

Comment:
probably the C++ template instantiation limit, which is more than anyone will ever type.